packages/framework/src/Component/EntityExtension/EntityExtensionSubscriber.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+namespace Shopsys\FrameworkBundle\Component\EntityExtension;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Configuration;
+use Doctrine\ORM\Event\LoadClassMetadataEventArgs;
+use Doctrine\ORM\Events;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use LogicException;
+use ReflectionClass;
+use Webmozart\Assert\Assert;
+
+/**
+ * Inspired by joschi127/doctrine-entity-override-bundle (https://github.com/joschi127/doctrine-entity-override-bundle)
+ */
+class EntityExtensionSubscriber implements EventSubscriber
+{
+    /**
+     * @var array<class-string, class-string>
+     */
+    protected array $entityExtensionMap;
+
+    /**
+     * @var bool
+     */
+    protected bool $isEntityMapLoaded = false;
+
+    /**
+     * @var array<class-string, array<class-string, class-string>>
+     */
+    protected array $parentClassesByClass = [];
+
+    /**
+     * @var array<class-string, class-string>
+     */
+    protected array $allParentClasses = [];
+
+    /**
+     * @var class-string[]|null
+     */
+    protected ?array $allRegisteredEntitiesCache = null;
+
+    /**
+     * @var \Doctrine\ORM\Configuration
+     */
+    protected Configuration $configuration;
+
+    /**
+     * @param array<class-string, class-string> $entityExtensionMap
+     */
+    public function __construct(array $entityExtensionMap)
+    {
+        $this->setEntityExtensionMap($entityExtensionMap);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getSubscribedEvents(): array
+    {
+        return [Events::loadClassMetadata];
```

Comment:
weird to see a constant that is not all caps :smile: 

I would like to take the opportunity here to pay homage to hip hop legend MF DOOM, aka Madvillain, RIP. https://www.youtube.com/watch?v=gSJeHDlhYls

Reply:
It is weird I agree :D 

Thanks for the sharing, cool ♫